### ISSUE TITLE
Sqlfluff with DBT seems to not lint compiled code.

### ISSUE THREAD

**Issue by RW-IliaGilmijarow** [sqlfluff/sqlfluff#5825]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When I have jinja code produce a very long sql line after compilation - this is **not** detected.

### Expected Behaviour

sqlfluff highlights and reports the *line_too_long* error for the long sql line

### Observed Behaviour

The long line is not reported

### How to reproduce

`SELECT
    '{{ "Lorem Ipsum Dolor " * 20 }}' AS my_long_line`

### Dialect

snowflake

### Version

Using SQLFluff v3.0.5.
dbt=1.7.13
Registered adapter: snowflake=1.7.3

### Configuration

.sqlfluff
```
[sqlfluff]
templater = dbt
dialect = snowflake
# This change (from jinja to dbt templater) will make linting slower
# because linting will first compile dbt code into data warehouse code.
runaway_limit = 10
max_line_length = 160
indent_unit = space

# exclude a no-no for USING syntax (https://docs.sqlfluff.com/en/stable/rules.html#rule-ST07)
exclude_rules = ST07

[sqlfluff:indentation]
tab_space_size = 4
indented_ctes = True

[sqlfluff:layout:type:comma]
spacing_before = touch
line_position = leading

[sqlfluff:rules:layout.indent]
hanging_indents = False

[sqlfluff:rules:capitalisation.keywords] 
capitalisation_policy = upper

[sqlfluff:rules:aliasing.table]
aliasing = explicit

[sqlfluff:rules:aliasing.column]
aliasing = explicit

[sqlfluff:rules:aliasing.expression]
allow_scalar = False

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

[sqlfluff:rules:ambiguous.column_references]  # Number in group by
group_by_and_order_by_style = implicit

[sqlfluff:rules:references.special_chars]
# Special Characters in Identifiers
allow_space_in_identifier = True
additional_allowed_characters = ['.','(',')','-']

[sqlfluff:rules:aliasing.length]
# Table Alias Lengths
min_alias_length = 3
```
we don't have a .sqlfluffignore

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by kzajaczkowski** [comment]
Hey @RW-IliaGilmijarow, did you check option `ignore_templated_areas = False`?

**Comment by RW-IliaGilmijarow** [comment]
> Hey @RW-IliaGilmijarow, did you check option `ignore_templated_areas = False`?

Now I did. It was not set. I tried setting it to True and False - neither one helped.

**Comment by RW-IliaGilmijarow** [comment]
@kzajaczkowski Sorry. I take that back. It does help. With this setting compiled SQL gets linted too.
I only realized today that my .sqlfluff file was in the wrong directory (not project root), and nothink I changed in it made any difference.

Now that I've moved the file the setting makes a difference.
